src/popup/components/Modals/PayloadForm.vue

```diff
@@ -0,0 +1,118 @@
+<template>
+  <Modal
+    from-bottom
+    no-padding
+    has-close-button
+    class="payload-form"
+    @close="resolve"
+  >
+    <div class="header">
+      {{ $t('modals.payload-form.title') }}
+    </div>
+    <div class="desc">
+      {{ $t('modals.payload-form.desc') }}
+    </div>
+    <div class="wrapper">
+      <label class="label">
+        {{ $t('modals.payload-form.label') }}
+      </label>
+      <FormTextarea
+        v-model="value"
+      />
+    </div>
+    <template #footer>
+      <BtnMain
+        class="btn-cancel"
+        variant="secondary"
+        @click="resolve()"
+      >
+        {{ $t('modals.payload-form.cancel-btn') }}
+      </BtnMain>
+      <BtnMain
+        :disabled="!value.length"
+        @click="onDone()"
+      >
+        {{ $t('modals.payload-form.done-btn') }}
+      </BtnMain>
+    </template>
+  </Modal>
+</template>
+
+<script>
+import { ref, defineComponent } from '@vue/composition-api';
+import Modal from '../Modal.vue';
+import FormTextarea from '../FormTextarea.vue';
+import BtnMain from '../buttons/BtnMain.vue';
+
+export default defineComponent({
+  name: 'PayloadForm',
+  components: { BtnMain, FormTextarea, Modal },
+  props: {
+    resolve: { type: Function, required: true },
+    reject: { type: Function, required: true },
+    onChange: { type: Function, required: true },
+    payload: { type: String, required: true },
+  },
+  setup(props) {
+    const value = ref(props.payload);
+
+    const onDone = () => {
```

Comment:
for me it would be better to name that function `submit` or something like that - I had to think for a while what is it for